bolt-taproot-gossip.md

```diff
@@ -0,0 +1,748 @@
+  in [BIP340][bip-340].
+* `partial_signature`: a 32-byte partial Musig2 signature as defined
+  in [BIP-MuSig2][bip-musig2].
+* `public_nonce`: a 66-byte public nonce as defined in [BIP-MuSig2][bip-musig2].
+
+## TLV Based Messages
+
+The initial set of Lightning Network messages consisted of fields that had to 
+be present. Later on, TLV encoding was introduced which provided a backward 
+compatible way to extend messages. To ensure that the new messages defined in 
+this document remain as future-proof as possible, the messages will be pure 
+TLV streams. 
+
+## Taproot Channel Proof and Verification
+
+Taproot channel funding transaction outputs will be SegWit V1 (P2TR) outputs of
+the following form:
+
+```
+OP_1 <taproot_output_key> 
+```
+
+where 
+```
+taproot_internal_key = MuSig2.KeyAgg(MuSig2.KeySort(bitcoin_key_1, bitcoin_key_2))
+
+taproot_output_key = taproot_internal_key + tagged_hash("TapTweak", taproot_internal_key) * G
+```
+
+`taproot_interal_key` is the aggregate key of `bitcoin_key_1` and
+`bitcoin_key_2` after first sorting the keys using the
+[MuSig2 KeySort][musig-keysort] algorithm and then running the
+[MuSig2 KeyAgg][musig-keyagg] algorithm.
+
+Then, to commit to spending the taproot output via the keyspend path, a
+[BIP86][bip86-tweak] tweak is added to the internal key to calculate the
+`taproot_output_key`.
+
+As with legacy channels, nodes will want to perform some checks before adding 
+an announced taproot channel to their routing graph:
+
+1. The funding output of the channel being advertised exists on-chain, is an
+   unspent UTXO and has a sufficient number of confirmations. To allow this 
+   check, the SCID of the channel will continue to be included in the channel
+   announcement.
+2. The funding transaction is sufficiently bound to the Lightning context. 
+   Nodes will do this by checking that they are able to derive the output key 
+   found on-chain by using the advertised `bitcoin_key_1` and `bitcoin_key_2` 
+   along with a [BIP86 tweak][bip86-tweak]. This provides a slightly weaker
+   binding to the LN context than legacy channels do but at least somewhat 
+   limits how the output can be spent due to the script-path being disabled.
+3. The owners of `bitcoin_key_1` and `bitcoin_key_2` agree to be associated with  
+   a channel owned by `node_1` and `node_2`.
+4. The owners of `node_ID_1` and `node_ID_2` agree to being associated with the 
+   output paying to `bitcoin_key_1` and `bitcoin_key_2`. 
+
+For legacy channels, these last two proofs are made possible by including four
+separate signatures in the `channel_announcement`. However, Schnorr signatures
+and the MuSig2 protocol allow us to now aggregate these four signatures into a
+single one. Verifiers will then be able to aggregate the four keys
+(`bitcoin_key_1`, `bitcoin_key_2`, `node_ID_1` and `node_ID_2`) using MuSig2 key
+aggregation and then they can do a single signature verification check instead
+of four individual checks.
```

Comment:
Can we weaken this (and future proof it) a little, by publishing only the taproot_output_key?  Sure, that means any taproot output can pretend to be a channel, but *this is also a feature*.

Reply:
he means that you can "announce" a channel that is not actually a channel.

Reply:
without checking `node_ID_1` or `node_ID_2` somehow?

If we do only publish `taproot_output_key` does that mean we may accept a phony channel announcement?

Reply:
Well, both node ids have to sign it.  But we don't care what the UTXO looks like internally, just that they prove they could spend it.

Reply:
responded [here](https://github.com/lightning/bolts/pull/1059/files#r1143349386)